File: Striver/Trees/Q02.py
# inorder traversal of a binary tree

from Node import Node

def inOrder(root):
    if not root:
        return
    inOrder(root.left)
    print(root.data, end=" ")
    inOrder(root.right)

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

inOrder(root)

File: Striver/Trees/Q02.java
package Striver.Trees;

public class Q02{
    // Inorder traversal

    public void inorder(Node root){
        if(root == null) return;
        inorder(root.left);
        System.out.print(root.data + " ");
        inorder(root.right);
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);
        root.right.left = new Node(6);
        root.right.right = new Node(7);

        Q02 obj = new Q02();
        obj.inorder(root);
    }
}